### PR TITLE
Update most libretro cores

### PR DESCRIPTION
**Pull request by kivutar**
This updates most of the cores except a few that were not building:

 - citra
 - sameboy

I'm building images with this change, I can tell so far that Generic and RPi2 are building.

I'm going to boot these images on a few computers and see if the cores are still working.

### CODE REVIEW

**Comment by ToKe79** [comment]
I got build errors with citra (not this PR, but generally) with newer versions. First it complained about cmake, but when I upgraded cmake, it complained about missing header:
```
 #include_next <stdlib.h>
               ^~~~~~~~~~
```
For sameboy we need to switch to "buildbot" branch. So just add `PKG_GIT_BRANCH="buildbot"` to `package.mk` of sameboy and run `./libretro-update.sh -p sameboy`.

File: packages/libretro/retroarch/package.mk [hunk 1]
```diff
@@ -19,7 +19,7 @@
 ################################################################################
 
 PKG_NAME="retroarch"
-PKG_VERSION="3a981e0"
+PKG_VERSION="b3d3cbd"
```
Comment (ToKe79) [inline packages/libretro/retroarch/package.mk]: I think we should go with https://github.com/libretro/RetroArch/commit/6fc6bfbb243de4da05a86d1edc3950815a964f1e, which is 1.7.1 release

**Comment by ToKe79** [comment]
My proposed sameboy changes are over at https://github.com/kivutar/Lakka-LibreELEC/pull/2.

**Comment by kivutar** [comment]
There are no real (including ARM) test phase in RA, so checking out the exact 1.7.1 tag doesn't ensure that it is bug free anyway.

**Comment by ToKe79** [comment]
Images for test:
http://nightly.builds.lakka.tv/special_builds/coreupdate01032018/

**Comment by kivutar** [comment]
I updated cores once again. I tested every core on an RPi2. All work except PPSSPP.

**Pull request merged**